pkg/controller/nodenetworkconfigurationpolicy/conditions/manager.go

```diff
@@ -0,0 +1,79 @@
+package conditions
+
+import (
+	"context"
+	"github.com/go-logr/logr"
+
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/retry"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+
+	nmstatev1alpha1 "github.com/nmstate/kubernetes-nmstate/pkg/apis/nmstate/v1alpha1"
+)
+
+var (
+	log = logf.Log.WithName("policy/conditions/manager")
+)
+
+type Manager struct {
+	client     client.Client
+	nodeName   string
+	policyName types.NamespacedName
+	logger     logr.Logger
+}
+
+func NewManager(client client.Client, nodeName string, policyName types.NamespacedName) Manager {
+	return Manager{
+		client:     client,
+		nodeName:   nodeName,
+		policyName: policyName,
+		logger:     log.WithValues("node", nodeName, "policy", policyName),
+	}
+}
+
+func (m *Manager) Progressing() {
```

Comment:
SetProgressing. As you said, Progressing would be a getter for the property. I think using Set in these methods would make the code little more readable. No questions to be asked.

Reply:
conditionsManager.EnactProgressing() ?

Reply:
I would prefer use "Notify" or "Enact" to notify events to the manager (is like a little fsm)